contracts/ProtocolReserve/ProtocolShareReserve.sol

```diff
@@ -148,61 +159,76 @@ contract ProtocolShareReserve is AccessControlledV8, IProtocolShareReserve {
     function setPrime(address _prime) external {
         _checkAccessAllowed("setPrime(address)");
         if (_prime == address(0)) revert InvalidAddress();
-        address oldPrime = prime;
+        emit PrimeUpdated(prime, _prime);
         prime = _prime;
-        emit PrimeUpdated(oldPrime, prime);
     }
 
     /**
      * @dev Add or update destination targets based on destination address
      * @param configs configurations of the destinations.
      */
-    function addOrUpdateDistributionConfigs(DistributionConfig[] memory configs) external {
-        _checkAccessAllowed("addOrUpdateDistributionConfigs(DistributionConfig)");
+    function addOrUpdateDistributionConfigs(DistributionConfig[] memory configs) external nonReentrant {
```

Comment:
Shouldn't we add the `nonReentrant` modifier also in the rest of public/external functions?

* setPoolRegistry
* setPrime
* updateAssetsState

Otherwise, for example, potentially someone could invoke `releaseFunds` and in the same execution re-enter into the contract invoking `updateAssetsState` (because it's not protected)